src/DefaultBuilder/src/WebHostEnvironment.cs

```diff
@@ -56,17 +56,15 @@ internal void ReadConfigurationSettings(IConfiguration configuration)
             WebRootPath = configuration[WebHostDefaults.WebRootKey] ?? WebRootPath;
         }
 
-        public void ApplyEnvironmentSettings(IWebHostBuilder genericWebHostBuilder)
+        public void ApplyEnvironmentSettings(IWebHostBuilder genericWebHostBuilder, HostBuilder hostBuilder)
         {
             genericWebHostBuilder.UseSetting(WebHostDefaults.ApplicationKey, ApplicationName);
             genericWebHostBuilder.UseSetting(WebHostDefaults.EnvironmentKey, EnvironmentName);
             genericWebHostBuilder.UseSetting(WebHostDefaults.ContentRootKey, ContentRootPath);
             genericWebHostBuilder.UseSetting(WebHostDefaults.WebRootKey, WebRootPath);
 
-            genericWebHostBuilder.ConfigureAppConfiguration((context, builder) =>
-            {
-                CopyPropertiesTo(context.HostingEnvironment);
-            });
+            var context = (WebHostBuilderContext)hostBuilder.Properties[typeof(WebHostBuilderContext)];
+            CopyPropertiesTo(context.HostingEnvironment);
```

Comment:
Do you even need to do this?

Reply:
If not, we get test failures [here](https://github.com/dotnet/aspnetcore/blob/dd476b1fdfd851dcfe0cab21c5084f05059de6bb/src/DefaultBuilder/test/Microsoft.AspNetCore.Tests/WebApplicationTests.cs#L189) and other similar places since WebApplication.Environment pulls from the IServiceProvider and we no longer register our IWebHostEnvironment as a service.